assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -275,43 +276,114 @@ export const selectors = {
 	 */
 	haveConversionEventsForTailoredMetrics: createRegistrySelector(
 		( select ) => ( state, useNewEvents ) => {
-			const leadRelatedMetrics = [
-				KM_ANALYTICS_TOP_PAGES_DRIVING_LEADS,
-				KM_ANALYTICS_TOP_CITIES_DRIVING_LEADS,
-				KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_LEADS,
-			];
-			const conversionEventWidgets = {
-				purchase: [
-					KM_ANALYTICS_TOP_CITIES_DRIVING_PURCHASES,
-					KM_ANALYTICS_TOP_DEVICE_DRIVING_PURCHASES,
-					KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_PURCHASES,
-				],
-				add_to_cart: [
-					KM_ANALYTICS_TOP_CITIES_DRIVING_ADD_TO_CART,
-					KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_ADD_TO_CART,
-				],
-				contact: leadRelatedMetrics,
-				submit_lead_form: leadRelatedMetrics,
-				generate_lead: leadRelatedMetrics,
-			};
-
-			const purposeTailoredMetrics = select(
-				CORE_USER
-			).getAnswerBasedMetrics( null, true );
-
 			const conversionReportingEventsChange = useNewEvents
 				? select(
 						MODULES_ANALYTICS_4
 				  ).getConversionReportingEventsChange()?.newEvents
 				: select( MODULES_ANALYTICS_4 ).getDetectedEvents();
 
-			return conversionReportingEventsChange?.some( ( event ) =>
+			const currentTailoredMetrics =
+				select( CORE_USER ).getAnswerBasedMetrics();
+
+			const tailoredMetricsWithNewEvents = select(
+				CORE_USER
+			).getAnswerBasedMetrics( null, conversionReportingEventsChange );
+
+			return tailoredMetricsWithNewEvents?.some(
+				( metric, index ) =>
+					metric !== currentTailoredMetrics?.[ index ]
+			);
+		}
+	),
+
+	/**
+	 * Checks if there are key metrics widgets that relly on the conversion events that have been lost.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} TRUE if current metrics are depending on the conversion events that have been lost, FALSE otherwise.
+	 */
+	haveLostEventsForCurrentMetrics: createRegistrySelector(
+		( select ) => () => {
```

Comment:
Since this selector is only conditional check for showing the banner in `ConversionReportingNotificationCTAWidget`, thorough tests are included for it, and there was no need to duplicate these tests in `assets/js/components/KeyMetrics/ConversionReportingNotificationCTAWidget.test.js`  as well, they are fully covered in `assets/js/modules/analytics-4/datastore/conversion-reporting.test.js`